Plugins/GitHub3/GitHub3Plugin.cs

```diff
@@ -146,6 +158,79 @@ public override void Register(IGitUICommands gitUiCommands)
             {
                 GitHub.setOAuth2Token(GitHubLoginInfo.OAuthToken);
             }
+
+            gitUiCommands.PreCommit += GitUiCommands_PreCommit;
+            gitUiCommands.PostCommit += GitUiCommands_PostCommit;
+        }
+
+        private void GitUiCommands_PreCommit(object? sender, GitUIEventArgs e)
+        {
+            if (!_issueCommitMessageHelperEnabled.ValueOrDefault(Settings))
+            {
+                return;
+            }
+
+            if (string.IsNullOrEmpty(GitHubLoginInfo.OAuthToken))
+            {
+                e.GitUICommands.AddCommitTemplate(_noTokenError.Text, () => string.Empty, Icon);
+                return;
+            }
+
+            IHostedRemote[] hostedRemotes = GetHostedRemotes().ToArray();
+            if (hostedRemotes.Length == 0)
+            {
+                return;
+            }
+
+            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            {
+                bool multipleRemotes = hostedRemotes.Length > 1;
+                IReadOnlyList<Issue> issues = _gitHub.GetAssignedIssues();
+
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                if (issues?.All(i => i.Number == 0) ?? true)
+                {
+                    e.GitUICommands.AddCommitTemplate(_noAssignedIssues.Text, () => string.Empty, Icon);
+                    return;
+                }
+
+                IEnumerable<Issue> recentUserIssues = issues.Where(i => i.Number != 0 && hostedRemotes.Any(r => r.Owner == i.Repository.Owner.Login && r.RemoteRepositoryName == i.Repository.Name))
```

Comment:
Any estimations on the timing here, or rather time to query GH?
(probably OK)

Reply:
I only have 1 issue for testing so it's pretty quick (around 2ms). But I don't expect it to be slow.
What is much slower is `GetAssignedIssues()` with 400ms and `GetHostedRemotes()` 46ms.

I have pushed changes to optimize more (As a result, I don't think it slow down the opening ).